fmlcore/src/main/java/net/minecraftforge/fml/CrashReportCallables.java

```diff
@@ -5,33 +5,93 @@
     private static final List<ISystemReportExtender> crashCallables = Collections.synchronizedList(new ArrayList<>());
 
+    /**
+     * Register a custom {@link ISystemReportExtender}
+     */
     public static void registerCrashCallable(ISystemReportExtender callable)
     {
         crashCallables.add(callable);
     }
 
-    public static void registerCrashCallable(String headerName, Supplier<String> reportGenerator) {
-        registerCrashCallable(new ISystemReportExtender() {
+    /**
+     * Register a {@link ISystemReportExtender system report extender} with the given header name and content
+     * generator, which will always be appended to the system report
+     * @param headerName The name of the system report entry
+     * @param reportGenerator The report generator to be called when a crash report is built
+     */
+    public static void registerCrashCallable(String headerName, Supplier<String> reportGenerator)
+    {
+        registerCrashCallable(new ISystemReportExtender()
+        {
+            @Override
+            public String getLabel()
+            {
+                return headerName;
+            }
+
+            @Override
+            public String get()
+            {
+                return reportGenerator.get();
+            }
+        });
+    }
+
+    /**
+     * Register a {@link ISystemReportExtender system report extender} with the given header name and content
+     * generator, which will only be appended to the system report when the given {@link BooleanSupplier} returns true
+     * @param headerName The name of the system report entry
+     * @param reportGenerator The report generator to be called when a crash report is built
+     * @param active The supplier of the flag to be checked when a crash report is built
+     */
+    public static void registerCrashCallable(String headerName, Supplier<String> reportGenerator, BooleanSupplier active)
+    {
+        registerCrashCallable(new ISystemReportExtender()
+        {
             @Override
-            public String getLabel() {
+            public String getLabel()
+            {
                 return headerName;
             }
+
             @Override
-            public String get() {
+            public String get()
+            {
                 return reportGenerator.get();
             }
+
+            @Override
+            public boolean isActive()
+            {
+                try
+                {
+                    return active.getAsBoolean();
+                }
+                catch (Throwable t)
+                {
+                    LOGGER.warn("CrashCallable '{}' threw an exception while checking the active flag, disabling", headerName, t);
+                    return false;
+                }
+            }
         });
     }
 
-    public static List<ISystemReportExtender> allCrashCallables() {
+    public static List<ISystemReportExtender> allCrashCallables()
```

Comment:
I'm pondering whether we should take this out of the public API (with `@ApiStatus.Internal`). I'll leave that decision to the core team for now.